konduit-serving-cli/src/main/java/ai/konduit/serving/cli/launcher/command/build/extension/ProfileCommand.java

```diff
@@ -0,0 +1,509 @@
+                        default:
+                            throw new IllegalStateException("Unsupported cuda version: " + cudaVersion);
+                    }
+                    if(cudaRedistPackage != null && !profile.additionalDependencies().contains(cudaRedistPackage)) {
+                        out.format("Adding cuda redist package: %s as an additional dependency. This will be " +
+                                        "downloaded and setup automatically on runtime konduit server start build.%n",
+                                cudaRedistPackage);
+                        List<String> additionalDependencies = new ArrayList<>(profile.additionalDependencies());
+                        additionalDependencies.add(cudaRedistPackage);
+                        profile.additionalDependencies(additionalDependencies);
+                    }
+                }
+            }
+        }
+
+        return profile;
+    }
+
+    public static boolean isProfileExists(String profileName) {
+        return getAllProfiles().containsKey(profileName);
+    }
+
+    public static Map<String, Profile> firstTimeProfilesSetup() {
+        log.error("Performing first time profiles setup.");
+
+        SystemInfo systemInfo = new SystemInfo();
+        CentralProcessor.ProcessorIdentifier processorIdentifier = systemInfo.getHardware().getProcessor().getProcessorIdentifier();
+        String cpuArch = processorIdentifier.getMicroarchitecture();
+
+        Map<String, Profile> profiles = new HashMap<>();
+        Profile cpuProfile = new Profile();
+        if(cpuArch != null) {
+            if (StringUtils.containsIgnoreCase(cpuArch, "arm")) {
+                if(processorIdentifier.isCpu64bit()) {
+                    cpuProfile.cpuArchitecture("arm64");
+                } else {
+                    cpuProfile.cpuArchitecture("armhf");
+                }
+            }
+        }
+
+        String cpuProfileName = "CPU";
+        profiles.put(cpuProfileName, cpuProfile);
+
+        log.error("Looking for CUDA compatible devices in the current system...");
+        List<GraphicsCard> nvidiaGraphicsCard = new ArrayList<>();
+        for(GraphicsCard graphicsCard : systemInfo.getHardware().getGraphicsCards()) {
+            String vendor = graphicsCard.getVendor();
+            if (vendor != null && StringUtils.containsIgnoreCase(vendor, "nvidia")) {
+                nvidiaGraphicsCard.add(graphicsCard);
+            }
+        }
+
+        if(!nvidiaGraphicsCard.isEmpty() && !cpuProfile.operatingSystem().equalsIgnoreCase("mac")) {
+            log.error("Found the following cuda compatible devices in the local system: {}", nvidiaGraphicsCard);
+
+            String defaultCudaVersion =  "CUDA_10.1";
+            Profile cudaProfile = new Profile(defaultCudaVersion,
+                    "x86", cpuProfile.operatingSystem(), Arrays.asList("HTTP", "GRPC"),
+                    Collections.singletonList("ch.qos.logback:logback-classic:1.2.3"));
```

Comment:
Again we can't/shouldn't assume x86 here. And let's not have these default values here.
Ideally they should be centralized - something like `Profile.withDefaults(name, arch, os)`